homeassistant/components/medcom_ble/sensor.py

```diff
@@ -0,0 +1,124 @@
+from homeassistant.helpers.update_coordinator import (
+    CoordinatorEntity,
+    DataUpdateCoordinator,
+)
+
+from .const import DOMAIN, UNIT_CPM
+
+_LOGGER = logging.getLogger(__name__)
+
+# Note: no medcom device supports battery readings at the moment, but this
+# could be the case in the future.
+SENSORS_MAPPING_TEMPLATE: dict[str, SensorEntityDescription] = {
+    "cpm": SensorEntityDescription(
+        key="cpm",
+        translation_key="cpm",
+        native_unit_of_measurement=UNIT_CPM,
+        state_class=SensorStateClass.MEASUREMENT,
+        icon="mdi:radioactive",
+    ),
+    "battery": SensorEntityDescription(
+        key="battery",
+        device_class=SensorDeviceClass.BATTERY,
+        native_unit_of_measurement=PERCENTAGE,
+        state_class=SensorStateClass.MEASUREMENT,
+        entity_category=EntityCategory.DIAGNOSTIC,
+    ),
+}
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: config_entries.ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up Medcom BLE radiation monitor sensors."""
+
+    coordinator: DataUpdateCoordinator[MedcomBleDevice] = hass.data[DOMAIN][
+        entry.entry_id
+    ]
+
+    # See the Airthings BLE integration: this template system
+    # lets us change some units if we want to. We could use this
+    # in the future to map CPMs to dose rate, but this depends
+    # on the device calibration - the device only tells us the CPM.
+    # Another approach would be to store the calibration value in
+    # Home assistant directly as a configuration.
+    sensors_mapping = SENSORS_MAPPING_TEMPLATE.copy()
+
+    entities = []
+    _LOGGER.debug("got sensors: %s", coordinator.data.sensors)
+    for sensor_type, sensor_value in coordinator.data.sensors.items():
+        if sensor_type not in sensors_mapping:
+            _LOGGER.debug(
+                "Unknown sensor type detected: %s, %s",
+                sensor_type,
+                sensor_value,
+            )
+            continue
+        entities.append(
+            MedcomSensor(coordinator, coordinator.data, sensors_mapping[sensor_type])
+        )
+
+    async_add_entities(entities)
+
+
+class MedcomSensor(
+    CoordinatorEntity[DataUpdateCoordinator[MedcomBleDevice]], SensorEntity
+):
+    """Medcom BLE radiation monitor sensors for the device."""
+
+    _attr_has_entity_name = True
+
+    def __init__(
+        self,
+        coordinator: DataUpdateCoordinator[MedcomBleDevice],
+        medcom_device: MedcomBleDevice,
+        entity_description: SensorEntityDescription,
+    ) -> None:
+        """Populate the medcom entity with relevant data."""
+        super().__init__(coordinator)
+        self.entity_description = entity_description
+
+        name = medcom_device.name
+        if identifier := medcom_device.identifier:
+            name += f"-{identifier}"
+
+        self._attr_unique_id = f"{name}-{entity_description.key}"
```

Comment:
Please use address instead of name

https://developers.home-assistant.io/docs/entity_registry_index?_highlight=unique&_highlight=id&_highlight=re#unique-id-requirements

Reply:
I assume this was copied from airthings-ble

https://github.com/home-assistant/core/pull/99832

Reply:
Yep, used airthings-ble as the reference. If only MAC addresses were really unique, eh :) OK so add a prefix like `medcom_{medcom_device.address}_{entity_description.key}` ?

Reply:
Don't use the name because if the name changes the unique id will change and than all the entities will get recreated.

You don't need to include the domain or platform because they are already considered for the unique id.

`{mac}_{key}` is a good stable unique id

Reply:
Updated! Is `self._attr_unique_id` supposed to end up being the entity's unique ID ? Or is it just one of the parameters used to determine the eventual unique_id ?

Reply:
The full unique id is handled by the entity registry https://github.com/home-assistant/core/blob/fe5eba9b31bb525f075edf046af8b3542db85e84/homeassistant/helpers/entity_registry.py#L423